src/ohri-form-context.tsx

```diff
@@ -26,6 +26,7 @@ export interface EncounterContext {
   encounterDate: Date;
   setEncounterDate(value: Date): void;
   initValues?: Record<string, any>;
+  setObsGroupCounter?: any;
```

Comment:
What happens if there's more than one obs group field?

Reply:
Each will be handled individually, although I am not quite sure I understand what this question asks.

Reply:
You are passing `setObsGroupCounter` on the entire form whether there's need to track count or not. In a scenario where this is being referenced more than one, it might bring a problem especially because you are tracking it at the form context level

Reply:
Oh, I see what you mean now. So yes the `setObsGroupCounter` will only update the state with objects of type repeatObsGroupCounter, see [here](https://arc.net/l/quote/iczqbmqa), this object will always include the field id for identification purposes, and in the event the obsGroup is duplicated it will always push the object with the most recent count to the front of the array.